finagle-base-http/src/main/scala/com/twitter/finagle/http/TraceInfo.scala

```diff
@@ -11,7 +11,53 @@ private object TraceInfo {
     while (iter.hasNext) headers -= iter.next()
   }
 
+  def convertB3Trace(request: Request) =
+    if (request.headerMap.contains(Header.TraceContext)) {
+      def handleSampled(headers: HeaderMap,value: String): Unit =
+        value match {
+          case "0" => headers.set(Header.Flags,"0")
+          case "d" => headers.set(Header.Flags,"1")
```

Comment:
don't recall if in scala case falls through, but debug implies sampled. so you could also set the sampled header here